manifest/v1alpha/slo/metrics_sumo_logic.go

```diff
@@ -136,21 +125,50 @@ var sumoLogicLogsTypeValidation = validation.New[SumoLogicMetric](
 		return m.Type != nil && *m.Type == sumoLogicTypeLogs
 	})
 
-var sumoLogicTimeSliceRegexp = regexp.MustCompile(`(?m)\stimeslice\s(\d+\w+)\s`)
+func validateSumoLogicTimeslice(query string) error {
+	timeslice, timesliceString, containsAlias, err := getTimeSliceFromSumoLogicQuery(query)
+	if err != nil {
+		return err
+	}
 
-func getTimeSliceFromSumoLogicQuery(query string) (time.Duration, error) {
-	subMatches := sumoLogicTimeSliceRegexp.FindAllStringSubmatch(query, 2)
-	if len(subMatches) != 1 {
-		return 0, fmt.Errorf("exactly one timeslice declaration is required in the query")
+	if seconds := int(timeslice.Seconds()); seconds != 15 && seconds != 30 && seconds != 60 ||
+		strings.HasPrefix(timesliceString, "0") || // Sumo Logic doesn't support leading zeros in query body
+		strings.HasPrefix(timesliceString, "+") ||
+		strings.HasPrefix(timesliceString, "-") ||
+		strings.HasSuffix(timesliceString, "ms") {
+		return errors.Errorf("timeslice value must be 15, 30, or 60 seconds, got: [%s]", timesliceString)
 	}
-	subMatch := subMatches[0]
-	if len(subMatch) != 2 {
-		return 0, fmt.Errorf("timeslice declaration must match regular expression: %s", sumoLogicTimeSliceRegexp)
+
+	if !containsAlias {
+		return errors.New("timeslice operator requires an n9_time alias")
 	}
+	return nil
+}
+
+func getTimeSliceFromSumoLogicQuery(query string) (
+	tsDuration time.Duration, durationString string, containsAlias bool, err error,
+) {
+	r := regexp.MustCompile(`\stimeslice\s([-+]?(\d+[a-z]+\s?)+)\s(?:as n9_time)?`)
+	matchResults := r.FindAllStringSubmatch(query, 2)
+	if len(matchResults) == 0 {
+		return 0, "", false, fmt.Errorf("query must contain a 'timeslice' operator")
+	}
+	if len(matchResults) > 1 {
+		return 0, "", false, fmt.Errorf("exactly one 'timeslice' declaration is required in the query")
+	}
+	submatches := matchResults[0]
+
+	if submatches[1] != submatches[2] {
+		return 0, "", false, fmt.Errorf("timeslice interval must be in a NumberUnit form - for example '30s'")
+	}
+
 	// https://help.sumologic.com/05Search/Search-Query-Language/Search-Operators/timeslice#syntax
-	timeslice, err := time.ParseDuration(subMatch[1])
+	durationString = strings.TrimSpace(submatches[1])
```

Comment:
[nit] We could, for example, create a map using enums for every part (`duration`, `contains`, and every other part) instead of using _magic_ numbers [forcing to do a mental map that `1` is _duration_, `2` is _contains_, etc.] - but it's only a nitpicked idea :peace_symbol: